game-core/build.gradle

```diff
@@ -7,29 +7,13 @@ plugins {
 
 description = 'TripleA is a free online turn based strategy game and board game engine, similar to such board games as Axis & Allies or Risk.'
 mainClassName = 'games.strategy.engine.framework.GameRunner'
+version = getEngineVersion()
```

Comment:
The extraction of `getEngineVersion` makes for a very clean update, I liked that it was very easy to re-use in another potential sub-project. While looking at version numbers, there were two things that came to mind:
1. Could the game-core project also use this same function?
2. Outside of the scope of this PR for a moment perhaps, but as a new artifact there is no strict requirement to have the version number in the artifact name. I noticed that the server does not have the version number built in, that was intentional when working with infrastructure scripting as the version number was already in the URL, and given the previous version may have already been downloaded, the extra information in the file name was ultimately just redundant/not helpful. I somewhat suspect when we update infrastructure, we'll find it could be a bit simpler to not have the version number in the jar file name. I'm happy for us to decide that a bit later when updating infrastructure scripting.

Reply:
Ugh, I was mistaken in the above in some places. Perhaps we should unify this behavior. It was intended to stop adding version number to the artifact name as it was redundant.

But, it looks like we add version number.. For local build it is one artifact that gets 'dev' instead of the full engine version number, which is inconsistent and implies we have two mechanisms for doing this given that in the final release we have the full version number: 
```
./game-core/build/artifacts:
triplea-1.9.0.0.dev-all_platforms.zip  TripleA_1.9.0.0.dev_unix.sh	      TripleA_1.9.0.0.dev_windows-64bit.exe
TripleA_1.9.0.0.dev_macos.dmg	       TripleA_1.9.0.0.dev_windows-32bit.exe

./lobby/build/artifacts:
triplea-dev-server.zip
```
I think overall in this case the inconsistency and multiple ways of solving the same problem is the biggest issue beyond a redundant version number. That would imply that as a fast-follow we should use this new `getEngineVersion` method in each of the projects that produce a build artifact, or alternatively decide to drop version number from the build artifact name consistently. Long term I think that would be the best answer as it would lead to simplifications throughout the build and deploy pipeline. Getting user buy-in to not have the version number in the artifact name would take some work, but from a TripleA admin perspective we don't want version number in artifact names (deployment downloads by URL, version number is already baked in; and because the same version may already have been downloaded a possibly failed previous attempt, the version number does not necessarily help as we have to clean up any existing files anyways).

Sorry for the long paragraph there, I would suggest we plan to unify all artifacts to have version number in the same manner as a fast-follow; in parallel give notice we intend to remove version number from the game-core artifact, and then remove the version number simplifying the (gradle+travis) build and (infrastructure) deployment scripts.

Reply:
I think we should leave the version number in the artifact names.  A lot of times when debugging something, I will download multiple portable builds.  Having to manually name them during each download or renaming files that have been auto-suffixed by my browser (e,g, `file(1)`, `file(2)`, etc.) is annoying.

I haven't done anything with infrastructure, but for Travis, having the version in the file name hasn't really been an issue.  Usually the script already has the version in an environment variable, or simply using a bash globber when it doesn't know the version works fine if we're looking for something in the local file system (e.g. `java -jar bin/triplea-*-server-all.jar`).